File: CSharp/Items/MSCE.ElementSetTool/DgnElementSetTool.cs
/*--------------------------------------------------------------------------------------+
|   $safeitemname$.cs
|
+--------------------------------------------------------------------------------------*/

#region Bentley Namespaces
using BDPN = Bentley.DgnPlatformNET;
using BG = Bentley.GeometryNET;
using BM = Bentley.MstnPlatformNET;
#endregion

namespace $rootnamespace$
{
    class $safeitemname$ : BDPN.DgnElementSetTool
    {
    public $safeitemname$(int toolID, int toolName) : base() { }

#region DgnElementSetTool Members
protected override void OnRestartTool()
        {
            InstallNewInstance();
        }

        public override BDPN.StatusInt OnElementModify(BDPN.Elements.Element element)
        {

            return BDPN.StatusInt.Success;
        }

    protected override bool OnResetButton(BDPN.DgnButtonEvent ev)
    {
        ExitTool();
        return true;
    }

    protected override void ExitTool()
    {
        base.ExitTool();
    }

protected override bool OnInstall()
{

    return true;
}

protected override void OnPostInstall()
{

    base.OnPostInstall();
}

protected override bool OnDataButton(BDPN.DgnButtonEvent ev)
{

    return true;
}

protected override void OnDynamicFrame(BDPN.DgnButtonEvent ev)
{

}


public void InstallNewInstance(string unparsed = "")
        {
            $safeitemname$ tool = new $safeitemname$(0, 0);
            tool.InstallTool();
        }
        #endregion
    }
}

File: Visual Basic/Items/MSCE.ElementSetTool/DgnElementSetTool.vb
'*--------------------------------------------------------------------------------------+
'   $safeitemname$.vb
'
'+--------------------------------------------------------------------------------------*/

#Region "Bentley Namespaces"
Imports BDPN = Bentley.DgnPlatformNET
Imports BG = Bentley.GeometryNET
Imports BM = Bentley.MstnPlatformNET
#End Region

Class $safeitemname$
		Inherits BDPN.DgnElementSetTool

    Public Sub New(ToolId As Integer, toolName As Integer)
        MyBase.New()

    End Sub

#Region "DgnElementSetTool Members"
    Protected Overrides Sub OnRestartTool()
        InstallNewInstance()
    End Sub

    Public Overrides Function OnElementModify(element As BDPN.Elements.Element) As BDPN.StatusInt

        Return BDPN.StatusInt.Success
    End Function

    Protected Overrides Function OnResetButton(ev As BDPN.DgnButtonEvent) As Boolean
        ExitTool()
        Return True
    End Function

    Protected Overrides Sub ExitTool()
        MyBase.ExitTool()
    End Sub

    Protected Overrides Sub OnDynamicFrame(ByVal ev As BDPN.DgnButtonEvent)
    End Sub

    Protected Overrides Function OnInstall() As Boolean

        Return True
    End Function

    Protected Overrides Sub OnPostInstall()
        MyBase.OnPostInstall()
    End Sub

    Public Sub InstallNewInstance(ByVal Optional unparsed As String = "")
        Dim tool As $safeitemname$ = New $safeitemname$(0, 0)
        tool.InstallTool()
    End Sub
#End Region

End Class